obspy/io/sh/evt.py

```diff
@@ -153,23 +128,26 @@ def _mags(obj, evid, stamag=False):
                 warn('invalid value for magnitude: %s (event id %s)'
                      % (magv, evid))
             else:
-                mag = StationMagnitude if stamag else Magnitude
-                mags.append(mag(magnitude_type=magtype2, mag=float(magv)))
+                magv = float(magv)
+                mag = (StationMagnitude(station_magnitude_type=magtype2,
+                                        mag=magv) if stamag else
+                       Magnitude(magnitude_type=magtype2, mag=magv))
+                mags.append(mag)
     if len(mags) == 1:
         pm = mags[0].resource_id
     return mags, pm
 
 
 def _seed_id_map(inventory=None, id_map=None, id_default='.{}..{}'):
-    ret = {}
     if id_map is None:
         id_map = {}
+    ret = copy(id_map)
```

Comment:
Looks like `id_map` needs to be a dictionary if provided, so you could use `id_map.copy()` to get a shallow copy as well?

Or, maybe you rather even want to go for `deepcopy` instead? Just to be 100% sure you don't get unwanted side-effects (although it seems you're only ever expecting strings, i.e. non-modifiable types, in there)